src/Analyser/NodeScopeResolver.php

```
@@ -5139,7 +5139,13 @@ private function processArgs(
 				$scopeToPass = $closureBindScope;
 			}
 
-			if ($parameter instanceof ExtendedParameterReflection) {
+			if (
+				$parameterType === null
+				|| $parameterType instanceof MixedType
+				|| $parameterType->isCallable()->no()
```

I think we need a new `TypeUtils` method. `findCallableType` perhaps. It'd return the type for `isCallable()->yes()`. Additionally it should recursively inspect unions for types like `?callable` or `?Closure`.

---

But I think this logic is wrong. What mainly has to be handled is `$parameterCallImmediately->maybe()`. That's where you need to ask about this. And also in the `else` section where it's `$callCallbackImmediately = $calleeReflection instanceof FunctionReflection;` right now.

---

This logic fails for a couple reasons:

1) If the parameter is `mixed` but marked with `@param-immediately-invoked-callable` it'd get silently ignored.
2) It'd still mark parameter as immediately invoked if the type is for example `string` (which is maybe-callable).

---

thanks for the in-detail feedback. there are a lot of things to consider, I did not see yet.
additionally https://github.com/phpstan/phpstan/issues/13331#issuecomment-3136382085 was useful.

I tried adding more tests and adding comments about expectations, of what I understood from the comments.

thank you